File: .sdk-repos/orchestration-cluster-api-js/examples/agent-definition.ts
// Compilable usage examples for agent definition operations.
// These examples are type-checked during build to guard against API regressions.

import { type AgentDefinitionKey, createCamundaClient } from '@camunda8/orchestration-cluster-api';

//#region GetAgentDefinition
async function getAgentDefinitionExample(agentDefinitionKey: AgentDefinitionKey) {
  const camunda = createCamundaClient();

  const definition = await camunda.getAgentDefinition(
    { agentDefinitionKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Name: ${definition.name}`);
  console.log(`Type: ${definition.agentType}`);
  console.log(`Element: ${definition.elementId}`);
}
//#endregion GetAgentDefinition

//#region SearchAgentDefinitions
async function searchAgentDefinitionsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchAgentDefinitions(
    {
      filter: { agentType: { $eq: 'AI_AGENT_TASK' } },
      sort: [{ field: 'name', order: 'ASC' }],
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const definition of result.items ?? []) {
    console.log(`${definition.agentDefinitionKey}: ${definition.name} (${definition.agentType})`);
  }
  console.log(`Total: ${result.page.totalItems}`);
}
//#endregion SearchAgentDefinitions

// Suppress "declared but never read"
void getAgentDefinitionExample;
void searchAgentDefinitionsExample;

File: .sdk-repos/orchestration-cluster-api-go/examples/agent_definition.go
// Agent definition operations: read and search.
package examples

import (
	"context"
	"fmt"

	camunda "github.com/camunda/orchestration-cluster-api-go"
	openapi "github.com/camunda/orchestration-cluster-api-go/client"
)

func getAgentDefinitionExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetAgentDefinition
	definition, err := client.GetAgentDefinition(ctx, openapi.MustAgentDefinitionKey("2251799813691958"))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", definition)
	// endregion GetAgentDefinition
	return nil
}

func searchAgentDefinitionsExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchAgentDefinitions
	result, err := client.SearchAgentDefinitions(ctx, *openapi.NewAgentDefinitionSearchQuery())
	if err != nil {
		return err
	}
	for _, d := range result.GetItems() {
		fmt.Printf("%v\n", d)
	}
	// endregion SearchAgentDefinitions
	return nil
}
